src/overload_hessian.jl

```diff
@@ -3,15 +3,29 @@
 function hessian_tracer_1_to_1(
     t::T, is_firstder_zero::Bool, is_secondder_zero::Bool
 ) where {T<:HessianTracer}
-    sg, sh = gradient(t), hessian(t)
-    sg_out, sh_out = hessian_tracer_1_to_1(sg, sh, is_firstder_zero, is_secondder_zero)
-    return T(sg_out, sh_out)
+    if t.isempty # TODO: add test
+        return t
+    else
+        pattern = hessian_tracer_1_to_1_pattern(
+            t.pattern, is_firstder_zero, is_secondder_zero
+        )
+        return T(pattern) # return tracer
+    end
 end
 
-function hessian_tracer_1_to_1(
+function hessian_tracer_1_to_1_pattern(
```

Comment:
New pattern types can be supported by adding `*_tracer_*_to_*_pattern` methods.

Reply:
Do we need the three layers `tracer -> pattern -> set`?

Reply:
It allows us to handle `isempty` on the tracer level instead of having to reimplement it for each pattern type.